src/encord_active/server/dependencies.py

```diff
@@ -28,9 +29,17 @@ async def get_project_file_structure(project: str) -> ProjectFileStructure:
                 status_code=status.HTTP_404_NOT_FOUND, detail=f"Project: {project} wasn't found in the DB"
             )
 
-        return ProjectFileStructure(
-            get_settings().SERVER_START_PATH / db_project.project_name.lower().replace(" ", "-")
-        )
+        base_path = get_settings().SERVER_START_PATH
+        direct_match = base_path / db_project.project_name.lower().replace(" ", "-")
+        if direct_match.is_dir():
+            return ProjectFileStructure(direct_match)
+
+        name_with_score = [
+            (path.name, SequenceMatcher(None, path.name, db_project.project_name).ratio())
+            for path in base_path.glob("*")
+        ]
+        most_similar, _ = sorted(name_with_score, key=lambda with_score: with_score[1], reverse=True)[0]
+        return ProjectFileStructure(get_settings().SERVER_START_PATH / most_similar)
```

Comment:
add a sanity check that project_hash in the yaml matches?

Reply:
I could just check the hash in the YAML file in the first place 🤷‍♂️